lib/google_search_data_viewer/keyword/keywords.ex

```diff
@@ -1,16 +1,21 @@
 defmodule GoogleSearchDataViewer.Keyword.Keywords do
   import Ecto.Query, warn: false
 
+  alias GoogleSearchDataViewer.Keyword.Queries.KeywordsQuery
   alias GoogleSearchDataViewer.Keyword.Schemas.KeywordUpload
   alias GoogleSearchDataViewer.Repo
 
   def get_keyword_upload(id), do: Repo.get(KeywordUpload, id)
 
+  def get_keyword_upload_with_search_results(keyword_upload_id) do
+    keyword_upload_id
+    |> KeywordsQuery.get_keyword_upload_with_search_results()
+    |> Repo.one()
+  end
```

Comment:
Should we reuse the `get_keyword_upload(id)` and `preload(:search_result_urls)` ?

I think `get_keyword_upload` returns only 1 item, so we can use `Repo.one!` instead of `Repo.get`

We can eliminate one method in the query `get_keyword_upload_with_search_results`

Reply:
Use `preload` in this function after calling `get_keyword_upload(id)`? Yeah think would be better

Reply:
yes, the idea is.

```elixir
  def get_keyword_upload(id), do: Repo.one!(KeywordUpload, id)

  def get_keyword_upload_with_search_results(keyword_upload_id) do
    keyword_upload_id
    |> get_keyword_upload()
    |> Repo.preload(:search_result_urls)
  end
```

I think i would be better because it eliminate the query method `KeywordsQuery.get_keyword_upload_with_search_results`